### PR TITLE
8328540: test javax/swing/JSplitPane/4885629/bug4885629.java fails on windows hidpi

### PR DESCRIPTION
**Pull request by prrace**
A simple fix to the test to skip sampling the leftmost and rightmost ends of the divider
to avoid possible issues at fractional scales.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8328540](https://bugs.openjdk.org/browse/JDK-8328540): test javax/swing/JSplitPane/4885629/bug4885629.java fails on windows hidpi (**Bug** - P4)

### Reviewers
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - Committer) ⚠️ Review applies to [9618fa4a](https://git.openjdk.org/jdk/pull/18384/files/9618fa4a8fa66fa34dbd00ccd7d9feee05cd5902)
 * [Prasanta Sadhukhan](https://openjdk.org/census#psadhukhan) (@prsadhuk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18384/head:pull/18384` \
`$ git checkout pull/18384`

Update a local copy of the PR: \
`$ git checkout pull/18384` \
`$ git pull https://git.openjdk.org/jdk.git pull/18384/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18384`

View PR using the GUI difftool: \
`$ git pr show -t 18384`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18384.diff">https://git.openjdk.org/jdk/pull/18384.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18384#issuecomment-2007857080)

### CODE REVIEW

**Review by kumarabhi006 — COMMENTED** [review]
1. should we mention `-Dsun.java2d.uiScale=2.0` in jtreg tag since it is supposed to be a hidpi test ?

2. Please update copyright year.

**Comment by prrace** [comment]
> 1. should we mention `-Dsun.java2d.uiScale` in jtreg tag since it is supposed to be a hidpi test ?

No, since it is not a hidpi test, it is just a test that happens to fail in certain hidpi envts.
And the fractional scale only works on windows anyway.
It might be "interesting" to run all our tests with  that option, but that's a different topic
and is more easily and robustly accomplished by running on an actual hidpi envt.

> 2. Please update copyright year.
OK ..

**Review by kumarabhi006 — APPROVED** [review]

File: test/jdk/javax/swing/JSplitPane/4885629/bug4885629.java [hunk 1]
```diff
@@ -104,9 +104,9 @@ public void run() {
 
                     SwingUtilities.convertPointToScreen(p, sp);
 
-                    for (int i = 0; i < rect.width; i++) {
+                    for (int i = 1; i < rect.width - 1; i++) {
                         if (!BGCOLOR.equals(robot.getPixelColor(p.x + i, p.y + rect.height - 1))) {
-                            throw new Error("The divider's area has incorrect color.");
+                            throw new Error("The divider's area has incorrect color. i=" + i);
```
Comment (prsadhuk) [inline test/jdk/javax/swing/JSplitPane/4885629/bug4885629.java:109]: Should we still use "Error" or use RuntimeException for consistency with other tests?

**Review by prsadhuk — APPROVED** [review]

**Pull request closed without merging**